#160702928 place order endpoint

nwashangai opened the pull request:
#### What does this PR do?
it adds place order functionality
#### Description of Task to be completed?
* create an endpoint to place an order
* validate endpoint
* write test cases to place an order
#### How should this be manually tested?
* Not applicable
#### Any background context you want to provide?
* None
#### What are the relevant pivotal tracker stories?
#160702928
#### Screenshots (if appropriate)
* None
#### Questions:
* None

nwashangai committed:
ft: place order
    - create endpoint for placing order
    - vaidate ndpoint
    - write test for place order endpoint
    [Finishes #160702928]

nwashangai committed:
ft: get order history for a particular user
    - crreate endpoint to get order history of user
    - write test for the endpoint
    [Finishes #160702946]

dogoyaro commented on src/controllers/OrderController.js:
```diff
@@ -1,21 +1,14 @@
-    } else {
-      response.status(400).json({ status: 'error', message: 'invalid user ID' });
-    }
+    OrderModel.makeOrder(request.body).then((result) => {;
```
What happens when the order placement fails?

dogoyaro commented on src/models/migration.js:
```diff
@@ -22,18 +22,40 @@ export default async () => {
         )`
-      ).catch(error => { throw error });
+      ).then(async () => {
+        await query(`INSERT INTO users(
```
Why are you programmatically creating this user with this credentials here?

nwashangai replied:
for test purposes

dogoyaro replied:
Don't do this here. Your tests should create the data themselves.

nwashangai replied:
but can I ignore the food menu creation until I create an endpoint for it because there is no endpoint for it?

dogoyaro replied:
I think you should do that.

dogoyaro commented on src/models/migration.js:
```diff
@@ -22,18 +22,40 @@ export default async () => {
         )`
-      ).catch(error => { throw error });
+      ).then(async () => {
+        await query(`INSERT INTO foods(
```
Same for this. I don't think your migration files should insert any data into your database.

nwashangai replied:
I thought you said you agreed I could leave this until the endpoint is created

dogoyaro commented on test/order.js:
```diff
@@ -8,86 +8,97 @@ chai.should();
         .send(order)
         .end((err, res) => {
-          res.should.have.status(200);
+          res.should.have.status(403);
```
Should this be a 403?

nwashangai committed:
Merge pull request #36 from nwashangai/ft-user-order-history-2197024

nwashangai committed:
ft: Admin can add food
    - create endpoint to add food
    - validate user inputs
    - write test cases for this endpoint
    [Finishes #160703460]

nwashangai committed:
ft: get food menu
    - create endpoint to get food menu
    - write test for this endpoint
    [Finishes #160703275]

nwashangai committed:
ft: update order status
    - create endpiont to update status
    - write test cases for this endpoint
    [Finishes #160703227]

nwashangai committed:
ft: admin gets specific order
    - create endpoint to get specific order
    - write test cases for this endpoint
    [Finishes #160703172]

nwashangai committed:
ft: admin gets all orders
    - create endpoint to get all orders
    - write test cases for this endpoint
    [Finishes #160703159]

nwashangai committed:
ft: adds update menu item functionalities
    - create endpoint to update menu item
    - write test cases for this endpoint
    [Finishes #160703481]

nwashangai committed:
Merge pull request #42 from nwashangai/ft-update-menu-item-2197024

#160703481 adds update menu item functionalities

nwashangai committed:
Merge pull request #41 from nwashangai/ft-get-all-orders-2197024

#160703159 admin gets all orders

nwashangai committed:
Merge pull request #40 from nwashangai/ft-get-specific-2197024

#160703172 admin gets specific order

nwashangai committed:
Merge pull request #39 from nwashangai/ft-update-status-2197024

nwashangai committed:
Merge pull request #38 from nwashangai/ft-get-menu-2197024

#160703275 get food menu

nwashangai committed:
Merge pull request #37 from nwashangai/ft-add-food-2197024
